scheduler/networktopology/network_topology.go

```diff
@@ -256,20 +295,39 @@ func (nt *networkTopology) DeleteHost(hostID string) error {
 		return err
 	}
 
+	for _, deleteKey := range deleteKeys {
+		nt.cache.Delete(deleteKey)
+	}
+
 	return nil
 }
 
 // Probes loads probes interface by source host id and destination host id.
 func (nt *networkTopology) Probes(srcHostID, destHostID string) Probes {
-	return NewProbes(nt.config.Probe, nt.rdb, srcHostID, destHostID)
+	return NewProbes(nt.config, nt.rdb, nt.cache, srcHostID, destHostID)
 }
 
 // ProbedCount is the number of times the host has been probed.
 func (nt *networkTopology) ProbedCount(hostID string) (uint64, error) {
 	ctx, cancel := context.WithTimeout(context.Background(), contextTimeout)
 	defer cancel()
 
-	return nt.rdb.Get(ctx, pkgredis.MakeProbedCountKeyInScheduler(hostID)).Uint64()
+	probedCountKey := pkgredis.MakeProbedCountKeyInScheduler(hostID)
+	if cache, ok := nt.cache.Get(probedCountKey); ok {
+		return cache.(uint64), nil
+	}
+
+	probedCount, err := nt.rdb.Get(ctx, probedCountKey).Uint64()
+	if err != nil {
+		return uint64(0), err
+	}
+
+	// Add cache data.
+	if err := nt.cache.Add(probedCountKey, probedCount, nt.config.Cache.TTL); err != nil {
```

Comment:
The cache already exists, why add it again? If you want to refresh expiration, please use `Set`.